write-fonts/src/tables/glyf.rs

```diff
@@ -25,6 +28,48 @@ pub struct SimpleGlyph {
     _instructions: Vec<u8>,
 }
 
+/// A glyph consisting of multiple component sub-glyphs
+#[derive(Clone, Debug)]
+pub struct CompositeGlyph {
+    bbox: Bbox,
+    components: Vec<Component>,
+    _instructions: Vec<u8>,
+}
+
+/// A single component glyph (part of a [`CompositeGlyph`]).
+#[derive(Clone, Debug)]
+pub struct Component {
+    glyph: GlyphId,
+    anchor: Anchor,
+    pub flags: ComponentFlags,
+    transform: Transform,
+}
```

Comment:
For reference, at time of writing fontmake-rs knows the glyph name - and could doubtless come up with a gid - and the transform. https://github.com/googlefonts/fontmake-rs/blob/main/fontir/src/ir.rs.